### PR TITLE
feat(backend): add listings closing routine

### PR DESCRIPTION
**Pull request by pbn4**
# Pull Request Template

## Issue Overview

This PR addresses #2193 

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

This changs adds NestJS Task scheduler to Listings service in order to run a listings status changing routine. 

## How Can This Be Tested/Reviewed?

Provide instructions so we can review.

Describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

**Comment by emilyjablonski** [comment]
How might we test this, or is this just a code review?

File: backend/core/src/listings/listings.service.ts [hunk 1]
```diff
@@ -212,4 +214,22 @@ export class ListingsService {
...
+    for (const listing of listings) {
+      if (listing.applicationDueDate && listing.applicationDueDate < new Date()) {
+        listing.status = ListingStatus.closed
+        shouldSave = true
+      }
+    }
+    if (shouldSave) {
+      await this.listingRepository.save(listings)
+    }
+  }
```
Comment (seanmalbert) [inline backend/core/src/listings/listings.service.ts:234]: Can you update this to query for listings where the application due date is in the past and the status is active and only update those.

Comment (seanmalbert) [reply backend/core/src/listings/listings.service.ts:234]: And with that update, update the tests to include active, pending and closed listings with past and present due dates and ensure only the applicable ones are updated.

**Review by seanmalbert — CHANGES_REQUESTED** [review]
@pbn4 ,
Looks good, just a minor update to `changeOverdueListingsStatusCron` to add a where clause and update the tests.

**Pull request merged**